components/announcement_bar/announcement_bar.jsx

```diff
@@ -312,27 +393,50 @@ export default class AnnouncementBar extends React.PureComponent {
                                 target='_blank'
                             >
                                 <FormattedMessage
-                                    id='error_bar.site_url.docsLink'
+                                    id='announcement_bar.error.site_url.docsLink'
                                     defaultMessage='Site URL'
                                 />
                             </a>
                         ),
                         link: (
                             <Link to='/admin_console/general/configuration'>
                                 <FormattedMessage
-                                    id='error_bar.site_url.link'
+                                    id='announcement_bar.error.site_url.link'
                                     defaultMessage='the System Console'
                                 />
                             </Link>
                         ),
                     }}
                 />
             );
+        } else if (message === AnnouncementBarMessages.EMAIL_VERIFICATION_REQUIRED) {
+            message = (
+                <React.Fragment>
+                    <FormattedHTMLMessage
+                        id={AnnouncementBarMessages.EMAIL_VERIFICATION_REQUIRED}
+                        defaultMessage='Check your email at {email} to verify the address. Cannot find the email? '
+                        values={{
+                            email: this.props.user.email,
+                        }}
+                    />
+                    {this.createEmailResendLink(this.props.user.email)}
+                </React.Fragment>
+            );
+        } else if (message === AnnouncementBarMessages.EMAIL_VERIFIED) {
+            message = (
+                <React.Fragment>
+                    <i className='fa fa-check'/>
+                    <FormattedHTMLMessage
+                        id={AnnouncementBarMessages.EMAIL_VERIFIED}
+                        defaultMessage=' Email verified'
```

Comment:
nit: `defaultMessage` has leading space?

Reply:
This one I kept so there is a space between the check mark and the text `Email verified`. Is there a better way to do this?

Reply:
Can we solve this with CSS? e.g. with the right constraints to isolate it to this component, write 
```css
.fa { margin-right: 1em; }
```

Alternatively -- though this is a last result -- you could add an explicit space in the code:
```js
<i className='fa fa-check'/>
{' '}
<FormattedHTMLMessage ... />
```

I raise this mostly because I think a leading or trailing space in a translation might get refactored out or even missed in translation to another language, and probably isn't the right way to encode this requirement.

Reply:
Oh yeah...that makes sense